From: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
Date: Tue, 1 Mar 2022 15:04:31 +0530
Subject: [PATCH] ath12k: fix QMI service arrive issue

QMI new server arrive callback not happen from the second insmod
of ath12k driver. It is due to unintentional register write happen
on PCIE_PCIE_CHDBOFF when configuring monitor ring setup. Since
mac type is not selected for monitor ring, it leads to UMAC ring
setup procedure for PMAC monitor rings. So assign the correct
mac type for monitor ring to avoid unintentional register write,
to resolve this issue.

Signed-off-by: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
---

--- a/drivers/net/wireless/ath/ath12k/hal.c
+++ b/drivers/net/wireless/ath/ath12k/hal.c
@@ -181,6 +181,7 @@ static const struct hal_srng_config hw_s
 		.start_ring_id = HAL_SRNG_SW2RXMON_BUF0,
 		.max_rings = 1,
 		.entry_size = sizeof(struct hal_mon_buf_ring) >> 2,
+		.mac_type = ATH12K_HAL_SRNG_PMAC,
 		.ring_dir = HAL_SRNG_DIR_SRC,
 		.max_size = HAL_RXDMA_RING_MAX_SIZE_BE,
 	},
@@ -188,6 +189,7 @@ static const struct hal_srng_config hw_s
 		.start_ring_id = HAL_SRNG_SW2TXMON_BUF0,
 		.max_rings = 1,
 		.entry_size = sizeof(struct hal_mon_buf_ring) >> 2,
+		.mac_type = ATH12K_HAL_SRNG_PMAC,
 		.ring_dir = HAL_SRNG_DIR_SRC,
 		.max_size = HAL_RXDMA_RING_MAX_SIZE_BE,
 	},
@@ -195,6 +197,7 @@ static const struct hal_srng_config hw_s
 		.start_ring_id = HAL_SRNG_RING_ID_WMAC1_RXMON2SW0,
 		.max_rings = 1,
 		.entry_size = sizeof(struct hal_mon_dest_desc) >> 2,
+		.mac_type = ATH12K_HAL_SRNG_PMAC,
 		.ring_dir = HAL_SRNG_DIR_DST,
 		.max_size = HAL_RXDMA_RING_MAX_SIZE_BE,
 	},
@@ -202,6 +205,7 @@ static const struct hal_srng_config hw_s
 		.start_ring_id = HAL_SRNG_RING_ID_WMAC1_TXMON2SW0_BUF0,
 		.max_rings = 1,
 		.entry_size = sizeof(struct hal_mon_dest_desc) >> 2,
+		.mac_type = ATH12K_HAL_SRNG_PMAC,
 		.ring_dir = HAL_SRNG_DIR_DST,
 		.max_size = HAL_RXDMA_RING_MAX_SIZE_BE,
 	}
